improver/generate_ancillaries/generate_timezone_mask.py

```diff
@@ -0,0 +1,342 @@
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module for generating timezone masks."""
+
+from datetime import datetime
+
+import iris
+import numpy as np
+import pytz
+from cf_units import Unit
+from iris.exceptions import CoordinateNotFoundError
+from pytz import timezone
+
+from improver import BasePlugin
+from improver.metadata.utilities import (
+    create_new_diagnostic_cube,
+    generate_mandatory_attributes,
+)
+from improver.utilities.cube_manipulation import collapsed
+from improver.utilities.spatial import lat_lon_determine, transform_grid_to_lat_lon
+
+
+class GenerateTimezoneMask(BasePlugin):
+
+    """
+    A plugin to create masks for regions of a geographic grid that are on different
+    timezones. The resulting masks can be used to isolate data for specific
+    timezones from a grid of the same shape as the cube used in this plugin
+    to generate the masks.
+    """
+
+    def __init__(self, ignore_dst=True, time=None, groupings=None):
+        """
+        Configure plugin options to generate the desired ancillary.
+
+        Args:
+            ignore_dst (Optional[bool]):
+                If True, find and use the UTC offset to a grid point ignoring
+                daylights savings.
+            time (Optional[str]):
+                A datetime specified in the format YYYYMMDDTHHMMZ at which to
+                calculate the mask (UTC). If daylights savings are ignored this
+                will have no impact on the resulting masks.
+            groupings (Optional[Dict[int, List[int, int]]]):
+                A dictionary specifying how timezones should be grouped if so
+                desired. This dictionary takes the form::
+
+                    {0: [-12, -5], 1:[-4, 4], 2: [5, 12]}
```

Comment:
Is this dictionary a correct example of a `groupings` dictionary, if the key is intended to be used as the point for the `UTC_offset` coordinate?

Reply:
Nope. Updated.